meteor/client/ui/SegmentTimeline/SegmentTimelineContainer.tsx

```diff
@@ -79,8 +77,90 @@ interface IProps extends IResolvedSegmentProps {
 	id: string
 }
 
-export const SegmentTimelineContainer = withResolvedSegment(
-	class SegmentTimelineContainer extends MeteorReactComponent<IProps & ITrackedProps, IState> {
+export function SegmentTimelineContainer(props: Readonly<IProps>): JSX.Element {
+	const partIds = useTracker(
+		() =>
+			Parts.find(
+				{
+					segmentId: props.segmentId,
+				},
+				{
+					fields: {
+						_id: 1,
+					},
+				}
+			).map((part) => part._id),
+		[props.segmentId],
+		[]
+	)
+	useSubscription(CorelibPubSub.pieces, [props.rundownId], partIds)
+
+	const partInstanceIds = useTracker(
+		() =>
+			PartInstances.find(
+				{
+					segmentId: props.segmentId,
+					reset: {
+						$ne: true,
+					},
+				},
+				{
+					fields: {
+						_id: 1,
+						part: 1,
+					},
+				}
+			).map((instance) => instance._id),
+		[props.segmentId]
+	)
+
+	const [debouncedPartInstanceIds, setDebouncedPartInstanceIds] = useState<PartInstanceId[] | undefined>(undefined)
+	useEffect(() => {
+		if (!partInstanceIds) return // Tracker is not ready yet
+
+		const timeout = setTimeout(() => {
+			setDebouncedPartInstanceIds((previousIds) => {
+				// // old subscription is equivalent to the new one, don't do anything
+				if (previousIds && equivalentArrays(previousIds, partInstanceIds)) {
+					return previousIds
+				} else {
+					return partInstanceIds
+				}
+			})
+		}, 40)
+
+		return () => {
+			// Abort if changed again before it fires
+			clearTimeout(timeout)
+		}
+	}, [partInstanceIds])
```

Comment:
We already have a `useDebounce` hook in `client/lib/lib.tsx`, maybe we could have an additional version of that hook with a comparator function (that this would feed `equivalentArrays` as), to shorten the body of this function?

Reply:
Good idea. This is done in 21f7bb00a202c5b8fd808ae47eaeebe11598b2b3